hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/test/GenericTestUtils.java

```diff
@@ -345,6 +356,33 @@ public static void assertExceptionContains(String expectedText,
     }
   }
 
+  /**
+   * Assert that an exception's <code>toString()</code> value
+   * matches the pattern.
+   * @param pattern regex pattern to match
+   * @param t thrown exception
+   * @param message any extra text for the string
+   * @throws AssertionError if the expected string is not found
+   */
+  public static void assertExceptionMatches(Pattern pattern,
+      Throwable t,
+      String message) {
+    assertNotNull(t, E_NULL_THROWABLE);
+    String msg = t.toString();
```

Comment:
rename to caughtString; as msg means "message" to me, and it isn't as the code doesn't use getMessage(). (which I'm happy with not using)